prime-router/src/testIntegration/kotlin/transport/SftpTransportIntegrationTests.kt

```diff
@@ -0,0 +1,478 @@
+            4,
+            "",
+            "",
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null
+        )
+        val contents = "HL7|Stuff"
+        val header = WorkflowEngine.Header(
+            task,
+            reportFile,
+            null,
+            settings.findOrganization("ignore"),
+            settings.findReceiver("ignore.SFTP_TEST"),
+            metadata.findSchema("covid-19"),
+            content = contents.toByteArray(),
+            true
+        )
+        val fileName = Report.formExternalFilename(header)
+        val actionHistory = ActionHistory(TaskAction.send)
+
+        val lsPath = "./path"
+        val remoteResourceInfos = listOf("a", "b", "c").map {
+            mockk<RemoteResourceInfo> {
+                every { this@mockk.toString() } returns "$it.txt"
+            }
+        }
+        val remoteFiles = listOf("a.txt", "b.txt", "c.txt")
+
+        val rmPath = "./path"
+        val rmFile = "a.txt"
+
+        val mockSSHClient = mockk<SSHClient>()
+        val mockSFTPClient = mockk<SFTPClient>()
+        val mockCredentialService = mockk<CredentialService>()
+    }
+
+    @BeforeEach
+    fun setUp() {
+        mockkObject(CredentialHelper)
+    }
+
+    @AfterEach
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun `happy path with user pass credentials`() {
+        val f = Fixture()
+
+        mockkObject(SftpTransport.Companion)
+
+        every { CredentialHelper.getCredentialService() } returns f.mockCredentialService
+        every {
+            f.mockCredentialService.fetchCredential(
+                "DEFAULT-SFTP",
+                "SftpTransport",
+                CredentialRequestReason.SFTP_UPLOAD
+            )
+        } returns UserPassCredential("foo", "pass")
+        every { SftpTransport.createDefaultSSHClient() } returns f.mockSSHClient
+        every { f.mockSSHClient.addHostKeyVerifier(any<HostKeyVerifier>()) } just runs
+        every { f.mockSSHClient.connect(f.transportType.host, f.transportType.port.toInt()) } just runs
+        every { f.mockSSHClient.authPassword("foo", "pass") } just runs
+        every { f.mockSSHClient.newSFTPClient() } returns f.mockSFTPClient
+        every { f.mockSFTPClient.fileTransfer.preserveAttributes = false } just runs
+        every { f.mockSFTPClient.put(any<LocalSourceFile>(), "${f.transportType.filePath}/${f.fileName}") } just runs
+        every { f.mockSFTPClient.close() } just runs
+        every { f.mockSSHClient.disconnect() } just runs
+
+        val retry = f.transport.send(
+            f.transportType,
+            f.header,
+            f.reportId,
+            null,
+            context,
+            f.actionHistory
+        )
+
+        assertThat(f.actionHistory.action.actionResult).startsWith("Success: sftp upload")
+        assertThat(retry).isNull()
+    }
+
```

Comment:
May we add "unhappy path ..." for each unit test.  Therefore, we can cover most of SFTPTransport code execution path.